File: ClientApp/src/app/models/dtos/MangaDto.ts
import { GenreType } from "../enums/GenreType";
import { StatusType } from "../enums/StatusType";

export class MangaDto {
    id: string = '';
    title: string = '';
    imgUrl: string = '';
    description: string = '';
    startDate: string |undefined;
    endDate: string |undefined;
    score: number = 0.00;
    status: StatusType = StatusType.Unknown;
    volume: number = 0;
    chapter: number = 0;
    rating: number = 0;
    author: string = '';
    adaptation: string = '';
    genre: Array<GenreType> = new Array<GenreType>;
}

File: MyListWebApplication/Models/DTOs/MangaDto.cs
using MyListWebApplication.Models.Enums;

namespace MyListWebApplication.Models.DTOs
{
    public class MangaDto
    {
        public string Id { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string ImgUrl { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public double Score { get; set; }
        public StatusType Status { get; set; }
        public int Volume { get; set; }
        public int Chapter { get; set; }
        public string Author { get; set; } = string.Empty;
        public int Rating { get; set; }
        public string Adaptation { get; set; } = string.Empty;
        public List<GenreType> Genre { get; set; } = [];
    }
}
